angularjs/src/lib/components/tokenfield/tokenfield.component.ts

```diff
@@ -76,18 +78,16 @@ export class TokenField implements ng.IDirective {
         .data('bs.tokenfield')
         .$input.on('paste', pasteEvent);
     });
-  };
 
-  public postLink: ng.IDirectiveLinkFn = (scope: ITokenField, elem: ng.IAugmentedJQuery, attr: ng.IAttributes) => {
     if (scope.static) {
       this.$timeout(() => {
-        let tokenField = elem.parent();
-        tokenField.addClass('form-control-static').removeClass('form-control');
-        let tokenInput = tokenField.find('input').last();
+        let tokenFieldElemParent = tokenfieldElem.parent();
+        tokenFieldElemParent.addClass('form-control-static').removeClass('form-control');
+        let tokenInput = tokenFieldElemParent.find('input').last();
         tokenInput.attr('readonly', 'true');
 
         // Make the input editable when token is being edited
-        elem
+        tokenfieldElem
```

Comment:
[Question] if the tokenfieldElem is changed here when removing readonly, is there any place else where this is added back (on a user action, fox example). If so, shouldn't tokenfieldElem be changed even there ?

Reply:
Mmm nope because the static setting is just an initial input value and not expression evaluated to change the component behavior when the input changes. Not modifying any other functionality here accept modifying the element references now that they're children of this component without using the `replace` option.